lib/features/contact/presentation/contact_controller.dart

```diff
@@ -162,14 +181,46 @@ class ContactController extends BaseController {
     }
   }
 
-  void selectContact(BuildContext context, EmailAddress emailAddress) {
-    KeyboardUtils.hideKeyboard(context);
-    popBack(result: emailAddress);
+  void handleOnSelectContactAction(EmailAddress emailAddress) {
+    log('ContactController::selectContact:emailAddress = $emailAddress');
+    final isEmailAddressExist = selectedContactList
+      .any((contact) => contact.emailAddress == emailAddress.emailAddress);
+    log('ContactController::selectContact:isEmailAddressExist = $isEmailAddressExist');
+    if (!isEmailAddressExist) {
```

Comment:
If the contact exists and user select it, should we remove the contact out of `selected`?

Reply:
No, we have a separate delete button for it.